src/dataflow/src/sink/kafka.rs

```diff
@@ -210,14 +255,16 @@ impl ProducerContext for SinkProducerContext {
 
     fn delivery(&self, result: &DeliveryResult, _: Self::DeliveryOpaque) {
         match result {
-            Ok(_) => (),
-            Err((e, msg)) => {
+            Ok(_) => self.retry_manager.lock().unwrap().record_success(),
```

Comment:
if you wanna avoid these unwraps, you can use `parking_lot` mutexes (but we should probably refactor and use them everywhere in the one big pr